packages/react/src/components/fileInput/FileInput.tsx

```diff
@@ -0,0 +1,365 @@
+  className?: string;
+  /**
+   * If `true`, the file input will be disabled
+   */
+  disabled?: boolean;
+  /**
+   * The error text content that will be shown below the input
+   */
+  errorText?: string;
+  /**
+   * The helper text content that will be shown below the input
+   */
+  helperText?: string;
+  /**
+   * If `true`, the label is displayed as required and the `input` element will be required
+   */
+  required?: boolean;
+  /**
+   * Override or extend the styles applied to the component
+   */
+  style?: React.CSSProperties;
+};
+
+const formatBytes = (bytes: number): string => {
+  if (bytes === 0) {
+    return '0 bytes';
+  }
+
+  const sizeUnits: string[] = ['B', 'kB', 'MB', 'GB', 'TB'];
+  const sizeUnitIndex = Math.floor(Math.log(bytes) / Math.log(1024));
+  const sizeInUnit = bytes / 1024 ** sizeUnitIndex;
+  return `${sizeUnitIndex < 2 ? Math.round(sizeInUnit) : sizeInUnit.toFixed(1)} ${sizeUnits[sizeUnitIndex]}`;
+};
+
+const getRemoveButtonLabel = (language: Language): string => {
+  return {
+    en: 'Remove',
+    fi: 'Poista',
+    sv: 'Ta bort',
+  }[language];
+};
+
+const getRemoveButtonAriaLabel = (language: Language, fileName: string): string => {
+  return {
+    en: `Remove ${fileName} from the added files list.`,
+    fi: `Poista tiedosto ${fileName} lisättyjen tiedostojen listasta.`,
+    sv: `Ta bort ${fileName} tillagda filen från fillistan.`,
+  }[language];
+};
+
+const getFileListAriaLabel = (language: Language): string => {
+  return {
+    en: 'Added files',
+    fi: 'Lisätyt tiedostot',
+    sv: 'Tillagda filerna',
+  }[language];
+};
+
+const getRemoveSuccessMessage = (language: Language): string => {
+  return {
+    en: 'File removed.',
+    fi: 'Tiedosto poistettu.',
+    sv: 'Filen borttagen.',
+  }[language];
+};
+
+const getAddSuccessMessage = (language: Language): string => {
+  return {
+    en: 'File added.',
+    fi: 'Tiedosto lisätty.',
+    sv: 'Filen har blivit tillagd.',
+  }[language];
+};
+
+export const FileInput = ({
+  id,
+  label,
+  buttonLabel,
+  language = 'en',
+  disabled,
+  dragAndDrop,
+  className = '',
+  errorText,
+  helperText,
+  onChange,
+  required,
+  style,
+  accept,
+  multiple,
+}: FileInputProps) => {
+  const inputRef = useRef<HTMLInputElement>(null);
+  const [selectedFiles, setSelectedFiles] = useState<File[]>([]);
+  const [successText, setSuccessText] = useState<string | undefined>();
+  const hasFilesSelected = selectedFiles && selectedFiles.length > 0;
+  const fileListId = `${id}-list`;
+  const fileListRef = useRef<HTMLUListElement>(null);
+  const fileListFocusIndexRef = useRef<number>();
+  const hasDragAndDrop = !!dragAndDrop;
+  const dropAreaRef = useRef<HTMLDivElement>(null);
+  const [isDragOverDrop, setIsDragOverDrop] = useState<boolean>(false);
+
+  const wrapperProps = {
+    className,
+    helperText,
+    successText,
+    errorText,
+    id,
+    label,
+    required,
+    style,
+  };
+
+  const passButtonClickToInput = () => {
```

Comment:
In single-file mode, should we warn the user that he is overwriting the previous file that he sent? That would probably require some kind of modal which might be tedious to implement here though 🤔

Reply:
Should we have a message which tells when file or files are replaced?

Reply:
Hmm.. that might be useful. Normally there's info that says 'Files added'. Maybe it could say: 'File replaced'? 🤔